flutter/lib/src/screenshot/recorder.dart

```diff
@@ -49,7 +58,10 @@ class ScreenshotRecorder {
       // On iOS, the screenshot resolution is not adjusted.
       final srcWidth = renderObject.size.width;
       final srcHeight = renderObject.size.height;
-      final pixelRatio = config.getPixelRatio(srcWidth, srcHeight);
+
+      final pixelRatio = config.getPixelRatio(srcWidth, srcHeight) ??
+          // ignore: deprecated_member_use
+          options.bindingUtils.instance!.window.devicePixelRatio;
```

Comment:
Maybe with this we could avoid deprecated code & non-null assertion:

```suggestion
          MediaQuery.of(context).devicePixelRatio;
```